solr/solrj/src/test/org/apache/solr/client/solrj/impl/BasicHttpSolrClientTest.java

```diff
@@ -341,7 +346,6 @@ public void testQuery() throws Exception {
       assertEquals("keep-alive", DebugServlet.headers.get("Connection"));
 
       // XML/POST
-      client.setParser(new XMLResponseParser());
```

Comment:
Is there a side effect here about having a new `XMLResponseParser` to be concerned with?

Reply:
i don't think so, as we generate a new client on line 236 with a fresh `XMLResponseParser`, and the test pasts.  However, the method `testUpdate` is giving me problems....

Reply:
so...   Reworked the tests...   still had to keep a `client.setParser()`, and never figured out why :-(